Merge scylla_install_(ami|image|azure) to single script

syuu1228 opened the pull request:
Deduplicate almost similar script for each cloud, and make it easier to
maintain.

syuu1228 commented:
Note: this doesn't including Azure support since #109 does not merged yet, but we will need to do it later.

syuu1228 commented:
Checked current scylla_install_azure implementation (https://github.com/scylladb/scylla-machine-image/blob/f587619a7a8686c03bf565fd07fbd5b318e1cf32/azure/image/files/scylla_install_azure), almost all difference comes from dropping CentOS image support, I think we can easily merge it to unified scylla_install_image script.

syuu1228 commented:
ping

penberg reviewed (dismissed):
Looks good to me.

syuu1228 commented:
Rebased with latest next, ping

syuu1228 commented:
Merged Azure support and rebased with latest next.

yaronkaikov commented:
@syuu1228 Is this patch ready?  if it is , let me modify PKG side before merging

syuu1228 commented:
@yaronkaikov yes it's ready

syuu1228 commented:
rebased with latest next.

yaronkaikov commented:
@syuu1228 I think you forgot to edit `aws/ami/build_deb_ami.sh`

yaronkaikov reviewed (changes requested):
Please also update `aws/ami/build_deb_ami.sh`

yaronkaikov commented on gce/image/build_image.sh:
```diff
@@ -195,6 +195,7 @@ echo "Calling Packer..."
   -var scylla_tools_version="$SCYLLA_TOOLS_VERSION" \
   -var scylla_python3_version="$SCYLLA_PYTHON3_VERSION" \
   -var scylla_build_id="$BUILD_ID" \
+  -var python="/usr/bin/python3" \
```
I think you can drop this change, we should remove this script as we build only for ubuntu based

hagitsegev commented:
@syuu1228 Can we go ahead with this?
I'm waiting for it to do something else.

syuu1228 commented:
@yaronkaikov @hagitsegev Dropped python variable changes, reverted to current code.

yaronkaikov commented on packer/scylla_install_image:
```diff
@@ -66,6 +66,7 @@ if __name__ == '__main__':
                         help='repository for install, specify .repo/.list file URL')
     parser.add_argument('--repo-for-update',
                         help='repository for update, specify .repo/.list file URL')
+    parser.add_argument('--target-cloud', choices=['aws', 'gce', 'azure'], help='specify target cloud')
```
@syuu1228 Don't we need to do this change for AWS as well? i mean use `--target-cloud` during ami creation ?

syuu1228 replied:
@yaronkaikov it specified at https://github.com/scylladb/scylla-machine-image/pull/132/files#diff-ad5a883a089226ec33413efcfae4cc5c88888c9ce2d631683675dde6c352dfb2R86

yaronkaikov commented on gce/image/scylla_gce.json:
```diff
@@ -38,9 +38,14 @@
       "source": "files/",
       "type": "file"
     },
+    {
```
I think that the lines above no longer exists , we should remove them, right? and i think it's the same for all cloud providers

syuu1228 replied:
@yaronkaikov Which lines you are mentioning about? "files/" ? If you are talking about "files/", it requires for --localrpm / --localdeb. If we only use --repo, we don't need it though.

yaronkaikov replied:
https://jenkins.scylladb.com/view/master/job/scylla-master/job/releng-testing/job/ami/651/console

syuu1228 replied:
@yaronkaikov Ah, okay I found the bug. When I removed scylla_install_* scripts from files/ directory, I didn't added .gitkeep file, so git thinks files/ directory no longer exists.
I pushed new version that can keep files/ directory.

syuu1228 commented:
Rebased with latest next, since it conflicted with 8915e3e40b377e3937c65fdcf773012fcf5cdb91

hagitsegev commented on packer/scylla_install_image:
```diff
@@ -114,37 +115,45 @@ if __name__ == '__main__':
+    if args.target_cloud == 'aws':
+        # use easy_install instead of pip, because easy_install requires
+        # fewer dependencies, no need to install gcc, it makes our AMI smaller.
+        if distro == 'centos':
```
We don't support CentOS based AMIs anymore. I think we can just drop this option.

syuu1228 replied:
@hagitsegev Can we just drop CentOS7 based AMI from next branch? If so, scripts become much simpler.
But I think it should be separated commit, separated PR, since it will be large changes
(we can drop half of scripts, drop scylla-machine-image rpm, drop support python2 in scylla_install_* script... etc)

yaronkaikov replied:
@syuu1228 I am not sure we should drop it yet, just today @hagitsegev had to build AMI based on Centos. let's keep it for a while and then do the cleanup

syuu1228 commented:
Rebased with latest next.

yaronkaikov reviewed (changes requested):
@syuu1228 AMI is now passing
Azure failed - https://jenkins.scylladb.com/view/master/job/scylla-master/job/releng-testing/job/azure-image/87/console
```
00:08:23  [1;31m==> azure-arm: userdel: user 'ubuntu' does not exist[0m
00:08:23  [1;31m==> azure-arm: Traceback (most recent call last):[0m
00:08:23  [1;31m==> azure-arm:   File "/home/azureuser/scylla_install_image", line 191, in <module>[0m
00:08:23  [1;31m==> azure-arm:     run('userdel -r -f {}'.format(distro))[0m
```
GCE failed as well - https://jenkins.scylladb.com/view/master/job/scylla-master/job/releng-testing/job/gce-image/400/console
```
00:03:50  [1;32m==> googlecompute: Uploading ../../packer/scylla_install_image => /home/ubuntu/[0m
00:03:50  [1;32m==> googlecompute: Provisioning with shell script: /tmp/packer-shell437782455[0m
00:03:51  [1;31m==> googlecompute: sudo: unable to execute /home/ubuntu/scylla_install_image: No such file or directory[0m
00:03:51  [1;31m==> googlecompute: Script exited with non-zero exit status: 1.Allowed exit codes are: [0][0m
```

You can rebuild those jobs to verify it's working or not , let me know if you need my assistance with testing it

syuu1228 commented:
> @syuu1228 AMI is now passing
> Azure failed - https://jenkins.scylladb.com/view/master/job/scylla-master/job/releng-testing/job/azure-image/87/console

Hmm, when I tested by myself, it just succeeded:
https://jenkins.scylladb.com/view/master/job/scylla-master/job/releng-testing/job/azure-image/91/console

syuu1228 commented:
> GCE failed as well - https://jenkins.scylladb.com/view/master/job/scylla-master/job/releng-testing/job/gce-image/400/console

Able to reproduce the issue on GCE and I'm guessing it is python path problem (the script starts with `#!/usr/bin/python` but there is no /usr/bin/python).
It is caused when we dropped ``` {{user `python`}} ``` on GCE.
To fix it we need to revert it or specify #!/usr/bin/python3.
I will add it again and update PR soon.

syuu1228 commented:
@yaronkaikov pushed fixed version.
Here's changes on new version:
```
diff --git a/azure/image/build_azure_image.sh b/azure/image/build_azure_image.sh
index 7525326..c70f74a 100755
--- a/azure/image/build_azure_image.sh
+++ b/azure/image/build_azure_image.sh
@@ -208,8 +208,7 @@ echo "Calling Packer..."
   -var client_secret="$AZURE_CLIENT_SECRET" \
   -var tenant_id="$AZURE_TENANT_ID" \
   -var subscription_id="$AZURE_SUBSCRIPTION_ID" \
-  -var scylla_build_id="$BUILD_ID" \
-  -var python="/usr/bin/python3" scylla_azure.json
+  -var scylla_build_id="$BUILD_ID" scylla_azure.json

   # For some errors packer gives a success status even if fails.
   # Search log for errors
diff --git a/azure/image/scylla_azure.json b/azure/image/scylla_azure.json
index f6d4c2b..44aaa13 100644
--- a/azure/image/scylla_azure.json
+++ b/azure/image/scylla_azure.json
@@ -40,7 +40,7 @@
     },
     {
       "inline": [
-        "sudo {{user `python`}} /home/{{user `ssh_username`}}/scylla_install_image --target-cloud azure {{ user `install_args` }}"
+        "sudo /usr/bin/python3 /home/{{user `ssh_username`}}/scylla_install_image --target-cloud azure {{ user `install_args` }}"
       ],
       "type": "shell"
     }
diff --git a/gce/image/build_deb_image.sh b/gce/image/build_deb_image.sh
index 796455c..84d31f4 100755
--- a/gce/image/build_deb_image.sh
+++ b/gce/image/build_deb_image.sh
@@ -206,6 +206,7 @@ echo "Calling Packer..."
   -var scylla_tools_version="$SCYLLA_TOOLS_VERSION" \
   -var scylla_python3_version="$SCYLLA_PYTHON3_VERSION" \
   -var scylla_build_id="$BUILD_ID" \
+  -var python=/usr/bin/python3 \
   scylla_gce.json

 # For some errors packer gives a success status even if fails.
diff --git a/gce/image/build_image.sh b/gce/image/build_image.sh
index 3bb5dd1..fb45e1f 100755
--- a/gce/image/build_image.sh
+++ b/gce/image/build_image.sh
@@ -195,6 +195,7 @@ echo "Calling Packer..."
   -var scylla_tools_version="$SCYLLA_TOOLS_VERSION" \
   -var scylla_python3_version="$SCYLLA_PYTHON3_VERSION" \
   -var scylla_build_id="$BUILD_ID" \
+  -var python=/usr/bin/python3 \
   scylla_gce.json

 # For some errors packer gives a success status even if fails.
diff --git a/gce/image/scylla_gce.json b/gce/image/scylla_gce.json
index 519ff51..5eafd6f 100644
--- a/gce/image/scylla_gce.json
+++ b/gce/image/scylla_gce.json
@@ -44,7 +44,7 @@
     },
     {
       "inline": [
-        "sudo /home/{{user `ssh_username`}}/scylla_install_image --target-cloud gce {{ user `install_args` }}"
+        "sudo {{user `python`}} /home/{{user `ssh_username`}}/scylla_install_image --target-cloud gce {{ user `install_args` }}"
       ],
       "type": "shell"
     }
@@ -57,6 +57,7 @@
     "image_storage_location": "",
     "instance_type": "",
     "ssh_username": "",
-    "source_image_family": ""
+    "source_image_family": "",
+    "python": ""
   }
 }
```

syuu1228 commented:
I found a better way to handle python path, run  update-alternatives prior to scylla_install_image, only when it's debian variants (= Ubuntu 20.04 on this script).
We don't need to use -var python on this way.
I will send new version of the PR after tested this.
```
diff --git a/aws/ami/scylla.json b/aws/ami/scylla.json
index 5315df1..d049a23 100644
--- a/aws/ami/scylla.json
+++ b/aws/ami/scylla.json
@@ -83,7 +83,8 @@
     {
       "inline": [
         "sudo /usr/bin/cloud-init status --wait",
-        "sudo {{user `python`}} /home/{{user `ssh_username`}}/scylla_install_image --target-cloud aws {{ user `install_args` }}"
+        "if [ -f /etc/debian_version ]; then sudo update-alternatives --install /usr/bin/python python /usr/bin/python3.8 1; fi",
+        "sudo /home/{{user `ssh_username`}}/scylla_install_image --target-cloud aws {{ user `install_args` }}"
       ],
       "type": "shell"
     }
@@ -99,8 +100,7 @@
     "security_group_id": "",
     "source_ami": "",
     "ssh_username": "",
-    "subnet_id": "",
-    "python": ""
+    "subnet_id": ""
   }
 }
```

syuu1228 commented:
> ```
> 00:08:23  �[1;31m==> azure-arm: userdel: user 'ubuntu' does not exist�[0m
> 00:08:23  �[1;31m==> azure-arm: Traceback (most recent call last):�[0m
> 00:08:23  �[1;31m==> azure-arm:   File "/home/azureuser/scylla_install_image", line 191, in <module>�[0m
> 00:08:23  �[1;31m==> azure-arm:     run('userdel -r -f {}'.format(distro))�[0m
> ```

I able to reproduced this error, will try to fix this too (seems like last time I tested wrong branch).

yaronkaikov commented:
> > ```
> > 00:08:23  �[1;31m==> azure-arm: userdel: user 'ubuntu' does not exist�[0m
> > 00:08:23  �[1;31m==> azure-arm: Traceback (most recent call last):�[0m
> > 00:08:23  �[1;31m==> azure-arm:   File "/home/azureuser/scylla_install_image", line 191, in <module>�[0m
> > 00:08:23  �[1;31m==> azure-arm:     run('userdel -r -f {}'.format(distro))�[0m
> > ```
> 
> I able to reproduced this error, will try to fix this too (seems like last time I tested wrong branch).

@syuu1228 Looks like AMI and GCE are now good. ping me when you fix the Azure so we can get it in

syuu1228 commented:
@yaronkaikov Pushed new version, changes are:
 - Fixed `userdel` error on Azure (stop running username configuration for Azure, since login user is not static just like GCE)
 - Applied `update-alternatives` to create /usr/bin/python symlink, dropped `-var python`

yaronkaikov commented:
@syuu1228 Azure still failing with the following error
```
08:23:51  [1;31m==> azure-arm: Traceback (most recent call last):[0m
08:23:51  [1;31m==> azure-arm:   File "/usr/lib/cnf-update-db", line 26, in <module>[0m
08:23:51  [1;31m==> azure-arm:     col.create(db)[0m
08:23:51  [1;31m==> azure-arm:   File "/usr/lib/python3/dist-packages/CommandNotFound/db/creator.py", line 94, in create[0m
08:23:51  [1;31m==> azure-arm:     self._fill_commands(con)[0m
08:23:51  [1;31m==> azure-arm:   File "/usr/lib/python3/dist-packages/CommandNotFound/db/creator.py", line 137, in _fill_commands[0m
08:23:51  [1;31m==> azure-arm:     with open(f) as fp:[0m
08:23:51  [1;31m==> azure-arm: FileNotFoundError: [Errno 2] No such file or directory: '/var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_focal_restricted_cnf_Commands-amd64'[0m
08:23:51  [0;32m    azure-arm: Fetched 19.8 MB in 3s (6238 kB/s)[0m
08:23:51  [0;32m    azure-arm: Reading package lists...[0m
08:23:51  [1;31m==> azure-arm: E: Problem executing scripts APT::Update::Post-Invoke-Success 'if /usr/bin/test -w /var/lib/command-not-found/ -a -e /usr/lib/cnf-update-db; then /usr/lib/cnf-update-db > /dev/null; fi'[0m
08:23:51  [1;31m==> azure-arm: E: Sub-process returned an error code[0m
08:23:51  [1;31m==> azure-arm: Traceback (most recent call last):[0m
08:23:51  [1;31m==> azure-arm:   File "/home/azureuser/scylla_install_image", line 107, in <module>[0m
08:23:51  [1;31m==> azure-arm:     run('apt-get update --allow-insecure-repositories -y')[0m
08:23:51  [1;31m==> azure-arm:   File "/home/azureuser/scylla_install_image", line 39, in run[0m
08:23:51  [1;31m==> azure-arm:     return subprocess.check_call(cmd, shell=shell, env=my_env)[0m
08:23:51  [1;31m==> azure-arm:   File "/usr/lib/python3.8/subprocess.py", line 364, in check_call[0m
08:23:51  [1;31m==> azure-arm:     raise CalledProcessError(retcode, cmd)[0m
08:23:51  [1;31m==> azure-arm: subprocess.CalledProcessError: Command '['apt-get', 'update', '--allow-insecure-repositories', '-y']' returned non-zero exit status 100.[0m
08:23:51  [1;31m==> azure-arm: Script exited with non-zero exit status: 1.Allowed exit codes are: [0][0m
08:23:51  [1;31m==> azure-arm: Step "StepProvision" failed, aborting...[0m
08:23:51  [1;31m==> azure-arm: aborted: skipping cleanup of step "StepConnectSSH"[0m
08:23:51  [1;31m==> azure-arm: aborted: skipping cleanup of step "StepGetIPAddress"[0m
08:23:51  [1;31m==> azure-arm: aborted: skipping cleanup of step "StepDeployTemplate"[0m
08:23:51  [1;31m==> azure-arm: aborted: skipping cleanup of step "StepValidateTemplate"[0m
08:23:51  [1;31m==> azure-arm: aborted: skipping cleanup of step "StepCreateResourceGroup"[0m
08:23:51  [1;31mBuild 'azure-arm' errored after 1 minute 24 seconds: Script exited with non-zero exit status: 1.Allowed exit codes are: [0][0m
```
https://jenkins.scylladb.com/view/master/job/scylla-master/job/releng-testing/job/azure-image/98/console

yaronkaikov commented:
Both AMI and GCE are passing : 
https://jenkins.scylladb.com/view/master/job/scylla-master/job/releng-testing/job/gce-image/421/ - GCE
https://jenkins.scylladb.com/view/master/job/scylla-master/job/releng-testing/job/ami/665/ - AMI
@syuu1228

syuu1228 commented:
> @syuu1228 Azure still failing with the following error
> 
> ```
> 08:23:51  �[1;31m==> azure-arm: Traceback (most recent call last):�[0m
> 08:23:51  �[1;31m==> azure-arm:   File "/usr/lib/cnf-update-db", line 26, in <module>�[0m
> 08:23:51  �[1;31m==> azure-arm:     col.create(db)�[0m
> 08:23:51  �[1;31m==> azure-arm:   File "/usr/lib/python3/dist-packages/CommandNotFound/db/creator.py", line 94, in create�[0m
> 08:23:51  �[1;31m==> azure-arm:     self._fill_commands(con)�[0m
> 08:23:51  �[1;31m==> azure-arm:   File "/usr/lib/python3/dist-packages/CommandNotFound/db/creator.py", line 137, in _fill_commands�[0m
> 08:23:51  �[1;31m==> azure-arm:     with open(f) as fp:�[0m
> 08:23:51  �[1;31m==> azure-arm: FileNotFoundError: [Errno 2] No such file or directory: '/var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_focal_restricted_cnf_Commands-amd64'�[0m
> 08:23:51  �[0;32m    azure-arm: Fetched 19.8 MB in 3s (6238 kB/s)�[0m
> 08:23:51  �[0;32m    azure-arm: Reading package lists...�[0m
> 08:23:51  �[1;31m==> azure-arm: E: Problem executing scripts APT::Update::Post-Invoke-Success 'if /usr/bin/test -w /var/lib/command-not-found/ -a -e /usr/lib/cnf-update-db; then /usr/lib/cnf-update-db > /dev/null; fi'�[0m
> 08:23:51  �[1;31m==> azure-arm: E: Sub-process returned an error code�[0m
> 08:23:51  �[1;31m==> azure-arm: Traceback (most recent call last):�[0m
> 08:23:51  �[1;31m==> azure-arm:   File "/home/azureuser/scylla_install_image", line 107, in <module>�[0m
> 08:23:51  �[1;31m==> azure-arm:     run('apt-get update --allow-insecure-repositories -y')�[0m
> 08:23:51  �[1;31m==> azure-arm:   File "/home/azureuser/scylla_install_image", line 39, in run�[0m
> 08:23:51  �[1;31m==> azure-arm:     return subprocess.check_call(cmd, shell=shell, env=my_env)�[0m
> 08:23:51  �[1;31m==> azure-arm:   File "/usr/lib/python3.8/subprocess.py", line 364, in check_call�[0m
> 08:23:51  �[1;31m==> azure-arm:     raise CalledProcessError(retcode, cmd)�[0m
> 08:23:51  �[1;31m==> azure-arm: subprocess.CalledProcessError: Command '['apt-get', 'update', '--allow-insecure-repositories', '-y']' returned non-zero exit status 100.�[0m
> 08:23:51  �[1;31m==> azure-arm: Script exited with non-zero exit status: 1.Allowed exit codes are: [0]�[0m
> 08:23:51  �[1;31m==> azure-arm: Step "StepProvision" failed, aborting...�[0m
> 08:23:51  �[1;31m==> azure-arm: aborted: skipping cleanup of step "StepConnectSSH"�[0m
> 08:23:51  �[1;31m==> azure-arm: aborted: skipping cleanup of step "StepGetIPAddress"�[0m
> 08:23:51  �[1;31m==> azure-arm: aborted: skipping cleanup of step "StepDeployTemplate"�[0m
> 08:23:51  �[1;31m==> azure-arm: aborted: skipping cleanup of step "StepValidateTemplate"�[0m
> 08:23:51  �[1;31m==> azure-arm: aborted: skipping cleanup of step "StepCreateResourceGroup"�[0m
> 08:23:51  �[1;31mBuild 'azure-arm' errored after 1 minute 24 seconds: Script exited with non-zero exit status: 1.Allowed exit codes are: [0]�[0m
> ```
> 
> https://jenkins.scylladb.com/view/master/job/scylla-master/job/releng-testing/job/azure-image/98/console

@yaronkaikov This is not reproducible for me, I re-run the job multiple time, it never occurred again.
In fact, the problem is not cloud specific thing, it should also occur on AMI and GCE but we got succeed.

Since FileNotFoundError is happened on apt cache file, it may networking problem on apt-get update, but normally we don't get such error message on networking problem, this can be something different.
Because the error is occurred on plug-in python script on APT::Update::Post-Invoke-Success handler, not at apt-get program itself.

One possible solution is removing `command-not-found` package, because the script never execute when the package not installed.
But maybe we don't have to remove it if the error never happens again.

yaronkaikov commented:
> > @syuu1228 Azure still failing with the following error
> > ```
> > 08:23:51  �[1;31m==> azure-arm: Traceback (most recent call last):�[0m
> > 08:23:51  �[1;31m==> azure-arm:   File "/usr/lib/cnf-update-db", line 26, in <module>�[0m
> > 08:23:51  �[1;31m==> azure-arm:     col.create(db)�[0m
> > 08:23:51  �[1;31m==> azure-arm:   File "/usr/lib/python3/dist-packages/CommandNotFound/db/creator.py", line 94, in create�[0m
> > 08:23:51  �[1;31m==> azure-arm:     self._fill_commands(con)�[0m
> > 08:23:51  �[1;31m==> azure-arm:   File "/usr/lib/python3/dist-packages/CommandNotFound/db/creator.py", line 137, in _fill_commands�[0m
> > 08:23:51  �[1;31m==> azure-arm:     with open(f) as fp:�[0m
> > 08:23:51  �[1;31m==> azure-arm: FileNotFoundError: [Errno 2] No such file or directory: '/var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_focal_restricted_cnf_Commands-amd64'�[0m
> > 08:23:51  �[0;32m    azure-arm: Fetched 19.8 MB in 3s (6238 kB/s)�[0m
> > 08:23:51  �[0;32m    azure-arm: Reading package lists...�[0m
> > 08:23:51  �[1;31m==> azure-arm: E: Problem executing scripts APT::Update::Post-Invoke-Success 'if /usr/bin/test -w /var/lib/command-not-found/ -a -e /usr/lib/cnf-update-db; then /usr/lib/cnf-update-db > /dev/null; fi'�[0m
> > 08:23:51  �[1;31m==> azure-arm: E: Sub-process returned an error code�[0m
> > 08:23:51  �[1;31m==> azure-arm: Traceback (most recent call last):�[0m
> > 08:23:51  �[1;31m==> azure-arm:   File "/home/azureuser/scylla_install_image", line 107, in <module>�[0m
> > 08:23:51  �[1;31m==> azure-arm:     run('apt-get update --allow-insecure-repositories -y')�[0m
> > 08:23:51  �[1;31m==> azure-arm:   File "/home/azureuser/scylla_install_image", line 39, in run�[0m
> > 08:23:51  �[1;31m==> azure-arm:     return subprocess.check_call(cmd, shell=shell, env=my_env)�[0m
> > 08:23:51  �[1;31m==> azure-arm:   File "/usr/lib/python3.8/subprocess.py", line 364, in check_call�[0m
> > 08:23:51  �[1;31m==> azure-arm:     raise CalledProcessError(retcode, cmd)�[0m
> > 08:23:51  �[1;31m==> azure-arm: subprocess.CalledProcessError: Command '['apt-get', 'update', '--allow-insecure-repositories', '-y']' returned non-zero exit status 100.�[0m
> > 08:23:51  �[1;31m==> azure-arm: Script exited with non-zero exit status: 1.Allowed exit codes are: [0]�[0m
> > 08:23:51  �[1;31m==> azure-arm: Step "StepProvision" failed, aborting...�[0m
> > 08:23:51  �[1;31m==> azure-arm: aborted: skipping cleanup of step "StepConnectSSH"�[0m
> > 08:23:51  �[1;31m==> azure-arm: aborted: skipping cleanup of step "StepGetIPAddress"�[0m
> > 08:23:51  �[1;31m==> azure-arm: aborted: skipping cleanup of step "StepDeployTemplate"�[0m
> > 08:23:51  �[1;31m==> azure-arm: aborted: skipping cleanup of step "StepValidateTemplate"�[0m
> > 08:23:51  �[1;31m==> azure-arm: aborted: skipping cleanup of step "StepCreateResourceGroup"�[0m
> > 08:23:51  �[1;31mBuild 'azure-arm' errored after 1 minute 24 seconds: Script exited with non-zero exit status: 1.Allowed exit codes are: [0]�[0m
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > https://jenkins.scylladb.com/view/master/job/scylla-master/job/releng-testing/job/azure-image/98/console
> 
> @yaronkaikov This is not reproducible for me, I re-run the job multiple time, it never occurred again.
> In fact, the problem is not cloud specific thing, it should also occur on AMI and GCE but we got succeed.
> 
> Since FileNotFoundError is happened on apt cache file, it may networking problem on apt-get update, but normally we don't get such error message on networking problem, this can be something different.
> Because the error is occurred on plug-in python script on APT::Update::Post-Invoke-Success handler, not at apt-get program itself.
> 
> One possible solution is removing `command-not-found` package, because the script never execute when the package not installed.
> But maybe we don't have to remove it if the error never happens again.

@syuu1228 Ok. so maybe it was something temporaty , make sense actually since you rebuild twice and it passed. 
Let's merge this

yaronkaikov reviewed (approved):
LGTM
